File: 0006-zigzag-conversion/0006-zigzag-conversion.cs
public class Solution {
    public string Convert(string s, int numRows) {
        if (numRows == 1){
            return s;
        }
        
        string[] rows = new string[numRows];
        bool down = true;
        int current = 0;
        
        for(int i = 0; i < s.Length; i++){
            rows[current] += s[i];
            if (down){
                current += 1;
            }
            else{
                current -= 1;
            }
            if (current == rows.Length){
                down = false;
                current -= 2;
            }else if(current == -1){
                down = true;
                current += 2;
            }
        }
        
    var result = string.Empty;
    foreach (var item in rows)
    {
        result += item;
    }
    return result;
    }
}

File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution:
    def convert(self, st: str, numRows: int) -> str:
        
        if numRows == 1:
            return st
    
        rows = [""] * numRows
        
        current = 0
        down = True
        for i, s in enumerate(st):
            rows[current] += s
            
            if down:
                current += 1 
            else:
                current -= 1
            
            if current == len(rows):
                down = False
                current -= 2
            elif current == -1:
                down = True
                current += 2
        return ''.join(rows)
            